frontend/packages/dev-console/src/components/health-checks/AddHealthCheckWrapper.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as yup from 'yup';
+import { Formik } from 'formik';
+import { FirehoseResult, LoadingBox, StatusBox, history } from '@console/internal/components/utils';
+import {
+  K8sResourceKind,
+  k8sUpdate,
+  modelFor,
+  referenceFor,
+  ContainerSpec,
+} from '@console/internal/module/k8s';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import { healthChecksProbesValidationSchema } from './health-check-probe-validation-utils';
+import { getHealthChecksData, getProbesData } from './create-health-check-probe-utils';
+import AddHealthCheck from './AddHealthCheck';
+import { HealthChecksProbeType } from './health-checks-types';
+
+type AddHealthCheckWrapperProps = {
+  resource?: FirehoseResult<K8sResourceKind>;
+  currentContainer: string;
+};
+
+const updateHealthChecksProbe = (
+  values,
+  resource: K8sResourceKind,
+  container: ContainerSpec[],
+): K8sResourceKind => {
+  const {
+    healthChecks: { readinessProbe, livenessProbe, startupProbe },
+    containerName,
+    healthChecks,
+  } = values;
+  const updatedResource = _.cloneDeep(resource);
+  const containerIndex = _.findIndex(updatedResource.spec.template.spec.containers, [
+    'name',
+    containerName,
+  ]);
+  updatedResource.spec.template.spec.containers[containerIndex] = {
+    ...container[containerIndex],
+    ...getProbesData(healthChecks, getResourcesType(resource)),
+  };
+
+  if (!readinessProbe.enabled) {
+    updatedResource.spec.template.spec.containers[containerIndex] = _.omit(
+      updatedResource.spec.template.spec.containers[containerIndex],
+      `${HealthChecksProbeType.ReadinessProbe}`,
+    );
+  }
+
+  if (!livenessProbe.enabled) {
+    updatedResource.spec.template.spec.containers[containerIndex] = _.omit(
+      updatedResource.spec.template.spec.containers[containerIndex],
+      `${HealthChecksProbeType.LivenessProbe}`,
+    );
+  }
+
+  if (!startupProbe.enabled) {
+    updatedResource.spec.template.spec.containers[containerIndex] = _.omit(
+      updatedResource.spec.template.spec.containers[containerIndex],
+      `${HealthChecksProbeType.StartupProbe}`,
+    );
+  }
+
+  return updatedResource;
+};
+
+const AddHealthCheckWrapper: React.FC<AddHealthCheckWrapperProps> = ({
```

Comment:
Calling things a `Wrapper` as a first class component is a poor name.
Perhaps simply `HealthChecksForm`.